buildingmotif/utils.py

```diff
@@ -242,10 +242,24 @@ def get_template_parts_from_shape(
             else:
                 param = _gensym()
             body.add((root_param, path, param))
-            deps.append({"template": str(otype), "args": {"name": param}})
-            body.add((param, RDF.type, otype))
 
-        # add 'hasValue'
+            # if otype is object of sh:class and it's also a shape, add it to the deps
+            if (None, SH["class"], otype) in shape_graph and (
+                otype,
+                RDF.type,
+                SH.NodeShape,
+            ) in shape_graph:
+                deps.append({"template": str(otype), "args": {"name": param}})
+                body.add((param, RDF.type, otype))
+
+            # if otype is a shape, add it to the deps
+            elif (otype, RDF.type, SH.NodeShape) in shape_graph or (
```

Comment:
couldn't you remove the if clause and just have this elif as the main logic? the elif logic is one component of the if anded logic, and the action is the same in both cases. So I think you can just get rid of that first block.

Reply:
Good catch! I've simplified this logic